app/routers/sessions.py

```diff
@@ -0,0 +1,81 @@
+from fastapi import APIRouter, status, HTTPException
+from fastapi.responses import JSONResponse
+from fastapi.encoders import jsonable_encoder
+from database import client
+from models import Session, SessionAnswer, SessionResponse
+
+router = APIRouter(prefix="/sessions", tags=["Sessions"])
+
+
+@router.post("/", response_model=SessionResponse)
+async def create_session(session: Session):
+    session = jsonable_encoder(session)
+
+    quiz = client.quiz.quizzes.find_one({"_id": session["quiz_id"]})
+
+    if quiz is None:
+        raise HTTPException(
+            status_code=404, detail=f"quiz {session['quiz_id']} not found"
+        )
+
+    previous_session = client.quiz.sessions.find_one(
+        {"quiz_id": session["quiz_id"], "user_id": session["user_id"]}
+    )
+
+    session_answers = []
+    if previous_session is None:
+        session["is_first"] = True
+
+        # since we know that there is going to be only one question set for now
+        if "question_sets" in quiz and quiz["question_sets"]:
+            question_set_id = quiz["question_sets"][0]["_id"]
+            questions = client.quiz.questions.find({"question_set_id": question_set_id})
+            if questions:
+
+                for question in questions:
+                    session_answers.append(
+                        jsonable_encoder(
+                            SessionAnswer.parse_obj(
+                                {
+                                    "question_id": question["_id"],
+                                }
+                            )
+                        )
+                    )
+    else:
+        session["is_first"] = False
+
+        # restore the answers from the previous sessions
+        session_answers = list(
+            client.quiz.session_answers.find({"session_id": previous_session["_id"]})
+        )
+        if session_answers:
+            for index, session_answer in enumerate(session_answers):
```

Comment:
No need of this if statement. The enumeration won't happen if session_answers is none or empty